ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -540,6 +549,43 @@ private EndpointResponse handleScalablePushQuery(
     return EndpointResponse.ok(queryStreamWriter);
   }
 
+  private EndpointResponse handleStreamPullQuery(
+      final ImmutableAnalysis analysis,
+      final ServiceContext serviceContext,
+      final ConfiguredStatement<Query> configured,
+      final CompletableFuture<Void> connectionClosedFuture) {
+
+    final TransientQueryMetadata transientQueryMetadata = ksqlEngine
+        .createStreamPullQuery(
+            serviceContext,
+            configured,
+            false
+        );
+    localCommands.ifPresent(lc -> lc.write(transientQueryMetadata));
```

Comment:
Nice, I think this is important.  This ensures that any transient queries that fail to get cleaned up will with a roll, and this is no different, so good to have it.

Reply:
Good to hear! I just copied it without understanding why.